gradle-conjure/src/main/java/com/palantir/gradle/conjure/ConjurePlugin.java

```diff
@@ -521,11 +542,14 @@ static void addGeneratedToMainSourceSet(Project subproj) {
         javaPlugin.getSourceSets().getByName("main").getJava().srcDir(subproj.files(JAVA_GENERATED_SOURCE_DIRNAME));
     }
 
-    static void applyDependencyForIdeTasks(Project project, Task compileConjure) {
+    static void applyDependencyForIdeTasks(Project project, TaskProvider<?> compileConjure) {
         project.getPlugins().withType(IdeaPlugin.class, plugin -> {
-            Task task = project.getTasks().findByName("ideaModule");
-            if (task != null) {
-                task.dependsOn(compileConjure);
+            // root project does not have the ideaModule task.  There is unfortunately no
+            // safe way to check for existence with the task avoidance APIs
+            try {
+                project.getTasks().named("ideaModule").configure(t -> t.dependsOn(compileConjure));
```

Comment:
From the [docs](https://docs.gradle.org/current/javadoc/org/gradle/api/tasks/TaskCollection.html#named-java.lang.String-):
> Locates a task by name, without triggering its creation or configuration, failing if there is no such object.

Could this end up in a race condition when the idea plugin sets up the task only after gradle-conjure is configured?

Reply:
Does this give us an advantage over `findByName`? I.e. `findByName` will also not create the task if it does not exist yet.

Reply:
but it will instantiate/initialize the task if it does exist, even if it's only been "registered" and not created. It's an annoying problem that there is no "is a TaskProvider with this name registered" call.